psi4/driver/procrouting/dft_funcs/dict_builder.py

```diff
@@ -0,0 +1,347 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2017 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This file is part of Psi4.
+#
+# Psi4 is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# Psi4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Psi4; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+"""
+Superfunctional builder function & handlers.
+The new definition of functionals is based on a dictionary with the following structure
+dict = {
+           "name":  "",       name of the functional - matched against name.upper() in method lookup
```

Comment:
Hopefully this is the right place to do it. Let me know if it should be moved.

Reply:
Probably as good of place as any. At some point we probably want to convert this to an actual [schema](http://json-schema.org) and go through and test these.

Reply:
Let me know your ideas. I think another candidate for `json`-ising is the dashparams dictionary, see below.